src/app/clusters/actions-server/actions-server.cpp

```diff
@@ -0,0 +1,418 @@
+            [this](HandlerContext & ctx, const auto & commandData) { HandleInstantActionWithTransition(ctx, commandData); });
+        return;
+    case Actions::Commands::StartAction::Id:
+        HandleCommand<Commands::StartAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleStartAction(ctx, commandData); });
+        return;
+    case Actions::Commands::StartActionWithDuration::Id:
+        HandleCommand<Commands::StartActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleStartActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::StopAction::Id:
+        HandleCommand<Commands::StopAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleStopAction(ctx, commandData); });
+        return;
+    case Actions::Commands::PauseAction::Id:
+        HandleCommand<Commands::PauseAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandlePauseAction(ctx, commandData); });
+        return;
+    case Actions::Commands::PauseActionWithDuration::Id:
+        HandleCommand<Commands::PauseActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandlePauseActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::ResumeAction::Id:
+        HandleCommand<Commands::ResumeAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleResumeAction(ctx, commandData); });
+        return;
+    case Actions::Commands::EnableAction::Id:
+        HandleCommand<Commands::EnableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleEnableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::EnableActionWithDuration::Id:
+        HandleCommand<Commands::EnableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleEnableActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableAction::Id:
+        HandleCommand<Commands::DisableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleDisableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableActionWithDuration::Id:
+        HandleCommand<Commands::DisableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleDisableActionWithDuration(ctx, commandData); });
+        return;
+    }
+}
+
+CHIP_ERROR ActionsServer::SetActionList(EndpointId aEndpoint, const ActionStructStorage & aAction)
```

Comment:
So this is not so much a SetActionList as a ModifyAction, no?

I am also a bit confused as to who will be calling this function.  We notify the delegate (i.e. the app) to update its state, but isn't it the app calling this function?

Reply:
A bridge can call this API to update an existing action in that case app needs to be notified.

Reply:
So two separate things:

1. The API is mis-named.
2. When you say "A bridge" and "app needs to be notified", what is "app" exactly?  Because in that case "app" is in fact "A bridge", no?  And if not, what is it?

Reply:
> The API is mis-named.

Changed the API name.

Reply:
You are right, a bridge is the app in this case. Removed the method that notifies the app.